src/schema/mutation/deleteLayer.mutation.ts

```diff
@@ -0,0 +1,43 @@
+import { GraphQLError, GraphQLNonNull, GraphQLID } from 'graphql';
+import { Layer } from '@models';
+import { LayerType } from '../../schema/types';
+import { AppAbility } from '@security/defineUserAbility';
+
+/**
+ * Edit new layer.
+ * Throw an error if user not connected and not permission to create.
+ */
+export default {
+  type: LayerType,
+  args: {
+    id: { type: new GraphQLNonNull(GraphQLID) },
+  },
+  async resolve(parent, args, context) {
+    const user = context.user;
+    if (!user) {
+      throw new GraphQLError(context.i18next.t('common.errors.userNotLogged'));
+    }
+
+    const layer = await Layer.findById(args.id);
+    const ability: AppAbility = user.ability;
+
+    if (ability.can('delete', layer)) {
+      const layers = await Layer.find({
```

Comment:
@ketanreliefapps can we put that in the methods that we execute when deleting models ?
check the application.model for example. There is a "addOnBeforeDeleteMany" where we tell what to do when we delete